### PR TITLE
Show pending transaction when contributing to GrantRound

### LINKED ISSUE #426: Donating to a GrantRound should give feedback and show pending
Currently when donating directly to a GrantRound there is no feedback and it may appear that nothing is happening.

- Implement a similar UX flow to what we have in the cart donate process where feedback is immediately given upon actions taken (like clicking donate)

### PR DESCRIPTION
**Pull request by mk1r**
resolves #426 

side note: found more tech-debt, and will add it to the tech-debt issue

### CODE REVIEW

File: app/src/views/GrantRound.vue [hunk 1]
```diff
@@ -44,7 +44,7 @@
         </InputRow>
 
         <!-- Submit and cancel buttons -->
-        <div class="flex justify-end pt-6">
+        <div v-if="roundStatus === ''" class="flex justify-end pt-6">
```
Comment (thelostone-mc) [inline app/src/views/GrantRound.vue:47]: Thoughts on swapping the if and else conditions.

```
<div v-if="roundStatus">...</div>
<div v-else>...</div>
```

File: app/src/views/GrantRound.vue [hunk 2]
```diff
@@ -329,19 +336,28 @@ function useGrantRoundDetail() {
     // contributionAmount must have the right number of decimals and be hexed
     const contributionAmount = parseUnits(amount, grantRound.value.matchingToken.decimals);
 
+    let txIndex = 1;
+    let txsNeeded = 1;
```
Comment (thelostone-mc) [inline app/src/views/GrantRound.vue:340]: is this needed ? it seems like it's always `txsNeeded = txIndex + 1`

File: app/src/views/GrantRound.vue (hunk 1 above)
Comment (mk1r) [reply app/src/views/GrantRound.vue:47]: was matching what was done in `cart.ts` but I like your suggestion

File: app/src/views/GrantRound.vue (hunk 2 above)
Comment (mk1r) [reply app/src/views/GrantRound.vue:340]: if you need to approve (line `343`) txsNeeded will be different

**Comment by apbendi** [comment]
hey @mk1r I'm getting the old BigNumber error, I think because this branch needs to be rebased. Can you do that real quick so we can evaluate this effectively?

**Comment by mk1r** [comment]
@apbendi sure, rebased

**Review by apbendi — APPROVED** [review]
Looks good, works well

**Review by thelostone-mc — APPROVED** [review]

**Pull request merged**